### PR TITLE
Added signature_type support

### PR DESCRIPTION
**Pull request by YusDyr**
I added initial support of signature repository.
I'm a newbie in ruby, so my code are far for ideal, I'm afraid, so check it before.

Usage example:
```
class repository {
    if ($operatingsystem == "FreeBSD") {
        if ($operatingsystemmajrelease * 1) == 10 {
        pkgng::repo { 'smart':
            packagehost => 'pkg.smartspb.net',
            repopath => '/packages/101amd64',
            protocol => "http",
            mirror_type => "http",
            pubkey => "/usr/local/etc/pkg/repos/smart.cert",
            priority => 100 }
        }
        file {'smart.cert':
                ensure  => file,
                path    => '/usr/local/etc/pkg/repos/smart.cert',
                owner   => root,
                group   => wheel,
                mode    => "0644",
                source  => "puppet:///files/others/pkg/smart.cert",
                purge   => true,
        }
    }
}
```
Result:

*cat /usr/local/etc/pkg/repos/smart.conf*
```
# File managed by Puppet

smart: {
    url:                "http://pkg.smartspb.net/packages/101amd64",
    mirror_type:        "http",
    enabled:            yes,
    priority:           100,
    pubkey:             "/usr/local/etc/pkg/repos/smart.cert",
    signature_type:     "pubkey"
}
```

### CODE REVIEW

File: templates/repo.erb [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 <%= @name -%>: {
-    url:         "<% if @mirror_type =~ /srv/i %>pkg+<% end %><%= @protocol %>://<%= @packagehost %><%= @repopath %>",
-    mirror_type: "<%= @mirror_type %>",
-    enabled:     <% if @enabled == true or @enabled == 'yes' %>yes<% else %>no<% end %>,
-    priority:    <%= @priority %>
+    url:		"<% if @mirror_type =~ /srv/i %>pkg+<% end %><%= @protocol %>://<%= @packagehost %><%= @repopath %>",
+    mirror_type:	"<%= @mirror_type %>",
+    enabled:		<% if @enabled == true or @enabled == 'yes' %>yes<% else %>no<% end %>,
+    priority:		<%= @priority %>,
+    <% if @pubkey != ''  %>pubkey:		"<%= @pubkey %>",
```
Comment (xaque208) [inline templates/repo.erb]: I'd prefer conditionals like this to be on their own line for readability.

**Comment by xaque208** [comment]
Hi.  Great, this looks fun.  The tests are failing for ruby 1.9, but I'll remove support later.  Would you mind squashing those commits into the first, and if its not too much trouble, changing the style on those conditionals to be on their own line.  The new line they add can be removed with a `-%>` using the minus at the end of the closing brace like that.

**Comment by YusDyr** [comment]
OK, I've done it and even slightly more )

12.01.2016 17:51, Zach Leslie пишет:
>
> Hi. Great, this looks fun. The tests are failing for ruby 1.9, but
> I'll remove support later. Would you mind squashing those commits into
> the first, and if its not too much trouble, changing the style on
> those conditionals to be on their own line. The new line they add can
> be removed with a |-%>| using the minus at the end of the closing
> brace like that.
>


-- 
Best regards,
Dennis Yusupoff

**Comment by YusDyr** [comment]
Any news?

**Comment by xaque208** [comment]
Well, I think a rebase is in order here as I merged some code ahead of this.  Apologies.  I'm still not sure why the tests failed on 1.9 though.  There should be tests for the repo work that at least let us know if this code will do the right thing.

**Comment by xaque208** [comment]
The spirit of this work has been integrated in 1.2.0.  Apologies that this took this long to get moving on.  I believe it covers all the cases you have outlined here.  Please have a test and let me know how it goes.

**Pull request closed without merging**